### PR TITLE
Update deployment to run migrations as a separate step

### PR DESCRIPTION
**Pull request by habdelra**
This PR adds support to run DB migrations as a separate step. In this PR I introduce a new Docker file that allows the migrations to run completely independent from the realm server (having a separate postgres workspace worked out really nicely). 

In this PR I actually make running the migrations automatically a flag that we set on the command line. Keep in mind that we want our tests to use the latest schema (the tests run in their own DBs). This also extends to matrix tests too. I opted to let the local dev version of the app auto migrate as well since that seems to work pretty well for our current dev experience. we can make that an opt-in feature if we want, but I'm guessing that most devs will want auto-migrate. it's one less thing to have to troubleshoot when things break (disabling auto migrate in your dev env is also quite simple). The hosted env's, though, do not auto migrate, and instead rely on the newly introduced migration deployment step for their migrations.

This PR depends on the infra PR https://github.com/cardstack/infra/pull/559 which provides an ECS service for the pg migration script to run in that is independent of the realm server.

As part of this I found a bunch of issues when we separated out the postgres workspace from the realm server in that specifically the migration related scripts in the realm-server workspace didn't work anymore. those actually needed to move over to the postgres workspace. I also had to tweak the `drop-all-dbs` to work using the new postgres workspace, which i renamed to `full-reset` to be more clear what it actually does.

TODO:
- [x] Deploy infra PR to staging
- [x] Deploy infra PR to production
- [x] Test deploy in staging

### CODE REVIEW

File: packages/realm-server/scripts/start-pg.sh
Comment (habdelra) [inline packages/realm-server/scripts/start-pg.sh:1]: it was so convenient having this script here I decided to just sym link it to its new location

File: packages/postgres/scripts/start-pg.sh
Comment (habdelra) [inline packages/postgres/scripts/start-pg.sh:1]: this is just a rename--the symlink i made to it confused github

File: packages/postgres/scripts/stop-pg.sh
Comment (habdelra) [inline packages/postgres/scripts/stop-pg.sh:1]: this is just a rename--the symlink i made to it confused github

**Review by backspace — APPROVED** [review]
Let’s see it in action!

**Comment by habdelra** [comment]
> Let’s see it in action!

infra has been deployed to staging and prod, and I confirmed the new boxel-pg-migrate ECS service is running. feel free to give it a shot. like i said, I'm not too sure that the workflow updates will be picked up outside of the `main` branch.

**Comment by backspace** [comment]
I think you’re right about this PR’s workflows not being used in deployment but I’ll try it to confirm.

**Comment by backspace** [comment]
It’s happening 😆 I’m honestly always unsure

boxel@f788e2b 2024-11-08 13-58-57

**Comment by habdelra** [comment]
Ah, I see an error in the docker image:
```
/bin/sh: 1: ./node_modules/.bin/node-pg-migrate: not found
```

**Comment by habdelra** [comment]
the migration script is now running succesfully in staging: 

image

**Review by lukemelia — APPROVED** [review]

**Pull request merged**